plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+
+            if hidden_field:
+                embeds = embeds or (discord.Embed(title='Tickets for {}'.format(userid)), )
+                embeds = (embed.add_field(name="Hidden", value=hidden_field) for embed in embeds)
+
+            if embeds:
+                await pager(msg.channel, *(Page(embed=embed) for embed in embeds))
+            else:
+                await reply("No tickets found for this user.")
+        elif isinstance(arg, commands.StringArg) and arg.text.isdigit():
+            # Assume provided number is a ticket id
+            if ticket := get_ticket(int(arg.text)):
+                await reply(embed=ticket.embed)
+            else:
+                await reply("No tickets found with this id!")
+    elif cmd == "showhidden":
+        """
+        Show hidden ticket(s) by ticketid or userid
+        """
+        arg = args.next_arg()
+        if isinstance(arg, commands.UserMentionArg):
+            # Collect hidden tickets for the mentioned user
+            userid = arg.id
+            tickets = fetch_tickets_where(status=TicketStatus.HIDDEN, targetid=userid)
+            embeds = summarise_tickets(
+                *tickets,
+                title='Hidden tickets for {}'.format(userid),
+                fmt="#{id}: **{type}** by <@{modid}>"
+            )
+
+            if embeds:
+                await pager(msg.channel, *(Page(embed=embed) for embed in embeds))
+            else:
+                await reply("No hidden tickets found for this user.")
+        elif isinstance(arg, commands.StringArg) and arg.text.isdigit():
+            # Assume provided number is a ticket id
+            if ticket := get_ticket(int(arg.text)):
+                await reply(embed=ticket.embed)
+            else:
+                await reply("No tickets found with this id!")
+    elif cmd == "history":
+        """
+        Show revision history for a given ticket
+        """
+        pass
+    else:
+        pass
+
+
+# ------------ Event handlers ------------
+
+util.discord.event("voice_state_update")(_read_audit_log)
+util.discord.event("member_ban")(_read_audit_log)
+util.discord.event("member_kick")(_read_audit_log)
+
+
+@util.discord.event("member_update")
+async def process_member_update(before, after):
+    if before.roles != after.roles:
+        await _read_audit_log()
+
+
+@util.discord.event("message")
+async def moderator_message(message):
+    if message.channel.type == discord.ChannelType.private and message.author.id in _ticketmods:
+        await _ticketmods[message.author.id].process_message(message)
+
+
+# Initial loading
+async def init_setup():
+    # Wait until the caches have been populated
+    await discord_client.client.wait_until_ready()
+
+    if not conf.guild or not discord_client.client.get_guild(conf.guild):
+        """
+        No guild, nothing we can do. Don't proceed with setup.
+        """
+        logger.critical("Guild not configured, or can't find the configured guild! Aborting setup.")
```

Comment:
Critical is a bit much. I'd go with error. How badly does the plugin break if this happens? I'd assume I can just reload the plugin.

Reply:
It's critical for the plugin since it can't do anything without a configured guild. Nothing in particular will break, it will just be non-functional, including mod reminders/ ticket expiry/ event handling.
It can just be reloaded, although I haven't yet tested integration with your reload system.

Reply:
yes but a `CRITICAL` loglevel in context of a program means it (the entire program) cannot function